ethereum-consensus/src/deneb/blob_sidecar.rs

```diff
@@ -27,6 +29,33 @@ pub struct BlobSidecar<
     pub kzg_commitment_inclusion_proof: Vector<Bytes32, KZG_COMMITMENT_INCLUSION_PROOF_DEPTH>,
 }
 
+// TODO: Convert method to free-standing function
+impl<const BYTES_PER_BLOB: usize, const KZG_COMMITMENT_INCLUSION_PROOF_DEPTH: usize>
+    BlobSidecar<BYTES_PER_BLOB, KZG_COMMITMENT_INCLUSION_PROOF_DEPTH>
+{
+    pub fn verify_blob_sidecar_inclusion_proof(&mut self) -> Result<(), Error> {
+        let blob_index = generalized_index_for_blob_index(self.index)?;
+        let subtree_index = get_subtree_index(blob_index);
+
+        let leaf = self.kzg_commitment.hash_tree_root()?;
+        let branch = self.kzg_commitment_inclusion_proof.as_ref();
+        let depth = KZG_COMMITMENT_INCLUSION_PROOF_DEPTH;
+        let root = self.signed_block_header.message.body_root;
+
+        Ok(is_valid_merkle_branch(leaf, branch, depth, subtree_index, root)?)
+    }
+}
+
+fn generalized_index_for_blob_index(i: usize) -> Result<GeneralizedIndex, MerkleizationError> {
+    let path = &["blob_kzg_commitments".into(), i.into()];
+    BeaconBlockBody::generalized_index(path)
+}
+
+fn get_subtree_index(i: GeneralizedIndex) -> usize {
+    let floorlog2 = i.checked_ilog2().expect("gindex isn't zero");
```

Comment:
we should prefer to return an error, rather than panicking, which `expect` would do here if `i == 0`

instead, we can do `unwrap_or` and return this error: https://github.com/ralexstokes/ssz-rs/blob/6329986d14e71a98a35cc9323f88cbef0332b6dd/ssz-rs/src/merkleization/mod.rs#L38

Reply:
`.unwrap_or()` expects a `u32` as input instead of an error.  I'm going explicitly handle the returned value of `.checked_ilog2()` with a `match` statement.  Thoughts?

Reply:
sorry I meant this: https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or